overlord/snapstate/handlers.go

```diff
@@ -960,12 +964,20 @@ func continueInhibitedAutoRefresh(st *state.State) {
 	st.EnsureBefore(0)
 }
 
-func abortMonitoring(st *state.State, snapName string) {
+func snapMonitoring(st *state.State) map[string]chan<- bool {
 	if cachedMonitored := st.Cached("monitored-snaps"); cachedMonitored != nil {
 		if monitoredSnaps, ok := cachedMonitored.(map[string]chan<- bool); ok {
-			monitoredSnaps[snapName] <- true
+			return monitoredSnaps
 		}
 	}
+
+	return nil
+}
+
+func abortMonitoring(st *state.State, snapName string) {
+	if monitored := snapMonitoring(st); monitored != nil {
+		monitored[snapName] <- true
```

Comment:
don't we need to check that monitored[snapName] is not nil? sending to nil channel panics

Reply:
Yes, I fixed it now and added a small test for it